Source/WTF/wtf/RAMSize.cpp

```diff
@@ -35,8 +35,13 @@
 #include <sys/sysinfo.h>
 #elif OS(UNIX)
 #include <unistd.h>
-#elif OS(MORPHOS) || OS(AMIGAOS)
+#elif OS(MORPHOS)
+#include <proto/exec.h>
+#include <exec/memory.h>
+#elif OS(AMIGAOS)
+#define __USE_INLINE__
```

Comment:
Can you give an example compilation error when you completely remove this and just add to the "CPPFLAGS" : -D__USE_INLINE__ universally?

Reply:
Here is a screenshot of the errors I get

![Screenshot from 2022-11-08 23-30-04](https://user-images.githubusercontent.com/14014909/200698274-063fbe13-bdb5-49bf-83da-f8163a50c5c4.png)

Reply:
I actually still do not understand how this works. For one thing, the screenshot shows a different compilation unit: "Lock.cpp", but this source code change is for a different unit: RAMSize.cpp which is possibly why I am getting confused.

Is it possible to just see what happens with "RAMSize.cpp" when you remove the #define / #undef and add in -D__USE_INLINE__. I want to see what happens in that case; what errors?

Reply:
If I recall correct, if I use -D__USE_INLINE__ it doesn't reach that point because of all the errors before that.

Reply:
Okay, that makes sense then.

The following:

1. #define __USE_INLINE__
2. #include <proto/exec.h>
3. #undef __USE_INLINE__

means that before line 1 there can never be any errors relating to using __USE_INLINE__ because it is not yet defined. At line 2 we include the header, which ends up including "inline4/exec.h" and defining all the old school Amiga functions to really go through the interface. Meaning that any calls like: "StackTrace()" - **even after the undef** will still continue to use the StackTrace() defined in the inline4 header file.

I am unsure as to why we need to undef __USE_INLINE__. I do not yet have access to actually inspect this for myself and compile it. The only reason to undef __USE_INLINE__ would be that any subsequent AmigaOS includes do not bring in their inline4 headers for old AmigaOS library calls.

I seem to dislike this approach more every time I look at it. Can we not remove the use of __USE_INLINE__ completely and instead convert to doing things through the library's interface?

Reply:
If we remove the `#undef __USE_INLINE__` then this will remain defined and possible we will have other parts of code breaking after that point. In my opinion, since `__USE_INLINE__` is not defined up to line 1, it should not be defined after line 2 as well.
The `__USE_INLINE__` is not necessary for every AmigaOS header file to be used. And remember, we try to do as less changes as possible, so as to keep that maintainable for the future.

Reply:
After a chat we decided that this is the lesser of all evils so far. I was under a bad impression that we could just swap MorphOS calls to IInterface->Method(), but we really do not want to do that to also avoid any potential conflicts when pulling changes into the Amiga branch. We want to make the changes as minimal as possible.

Let's allow this PR and think about any neater solutions in the future. All of these includes wrapped with def/undef __USE_INLINE__ can easily be grep-d for anyway!